### ISSUE TITLE
[mqtt] Homie devices not updating

### ISSUE THREAD

**Issue by davidgraeff** [openhab/openhab-addons#4886]
See also PR on https://github.com/pavax/openhab2-addons/pull/1

**Comment by boc-tothefuture** [comment]
@davidgraeff Does this fix the issue I [raised here](https://community.openhab.org/t/mqtt2-homie-convention-thing-goes-offline-permanently/63163/4?u=broconne).  That issue isn't resolved in 2.5M1 (just checked), I am happy to open another issue but didn't want to open another issue if its resolved here.

**Comment by davidgraeff** [comment]
@boc-tothefuture This is only the issue. There was a PR that made Homie devices update on an OH restart. So it might be fixed in a snapshot. But I'll do OH development only after I have finished my Paper UI replacement and when the IDE situation has stabilized.

**Comment by p0lyg0ne** [comment]
I also have the issue with OH 2.4.0.
In my case I can see that the device becomes online, but the properties are not updated.
Subscring to homie device topics via generic mqtt thing works.

**Comment by bodiroga** [comment]
We should wait until openHAB 2.5M2 version is released and then analyze all remaining bugs, to try to fix them before 2.5 is final :cry: Let's also hope that the IDE setup process will be ready soon :+1:

**Comment by tantive** [comment]
I can confirm the non-updating properties in OH 2.4.0.

**Comment by J-N-K** [comment]
Please provide logs and describe exactly what is working and what is not. What are „properties“? I‘m not familiar with homie-speech.

**Comment by davidgraeff** [comment]
@tantive We do not accept bug reports for OH 2.4. openHAB does not backport patches to specific versions.

**Comment by teliot** [comment]
still an issue.

when i first add the homie 3.0 thing in OH 2.5M1 many more channels are registered as subscribed that after restart. I have restarted the homie device but it does not help.

![Screenshot (1)](https://user-images.githubusercontent.com/6187302/62262886-24e37780-b3cf-11e9-9513-ab7edc355ade.png)

![Screenshot (2)](https://user-images.githubusercontent.com/6187302/62262898-2876fe80-b3cf-11e9-8113-d80aa6cab896.png)

**Comment by bodiroga** [comment]
@teliot, I'm not sure what those images should reflect, I don't see anything strange. What should we focus on?

**Comment by teliot** [comment]
This is for OH 2.5M1 i am looking for documentation on where to put the latest openhab-addons jar to test if still a bug.

The first picture shows all of the MQTT subs by OH when the thing is first created.
The second picture shows all of the OH subs after OH is restarted. In the second screenshot OH did not sub to temperature the channels; which is why its not updating any data for the thing.

i am wanting to try main latest and pavax#1 linked above but still learning OH and i don't know where to put the jar file (Ubuntu with bintree install).

**Comment by J-N-K** [comment]
Please check if this is still a problem in latest snapshot. At least the changes mentioned in the first post are already part of the codebase.

**Comment by teliot** [comment]
No change.
This is what i have on a fresh LXC 

apt list --installed | grep openhab
openhab2/now 2.5.0~M1-1 all [installed,upgradable to: 2.5.0~S1652-1]
openhab2-addons/unstable,now 2.5.0~S1652-1 all [installed]

It works until restart, then OH behaves the same as the screenshots above.

As far as i can tell the change mentioned in the first post was not merged.

**Comment by J-N-K** [comment]
Please show `bundle:list | grep -i mqtt` from the karaf/openhab console.

**Comment by teliot** [comment]
```
openhab> bundle:list | grep -i mqtt
198 x Active   x  80 x 1.2.0                  x Paho MQTT Client
200 x Active   x  80 x 0.11.0.oh250M1         x Eclipse SmartHome MQTT Binding
201 x Active   x  80 x 0.11.0.oh250M1         x Eclipse SmartHome MQTT Thing Binding
202 x Active   x  80 x 0.11.0.oh250M1         x Eclipse SmartHome MQTT Transport Bundle
openhab>
```


the homie convention is not coming up with that expression under this new version.
```
196 │ Active   │  80 │ 2.5.0.M1               │ HomeBuilder User Interface
197 │ Resolved │  75 │ 2.5.0.M1               │ openHAB Paper UI Theme Fragment, Hosts: 193
198 │ Active   │  80 │ 1.2.0                  │ Paho MQTT Client
199 │ Active   │  80 │ 0.11.0.oh250M1         │ Eclipse SmartHome Automation API
200 │ Active   │  80 │ 0.11.0.oh250M1         │ Eclipse SmartHome MQTT Binding
201 │ Active   │  80 │ 0.11.0.oh250M1         │ Eclipse SmartHome MQTT Thing Binding
202 │ Active   │  80 │ 0.11.0.oh250M1         │ Eclipse SmartHome MQTT Transport Bundle
openhab> 
```

While i was in this container i did apt update and saw a new version. I upgraded only the openhab2-addons to latest. However i saw no difference with bundle:list on these addons, Is there an additional step after an apt upgrade?

```
root@openhab25:~# apt list --installed | grep openhab

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

openhab2/now 2.5.0~M1-1 all [installed,upgradable to: 2.5.0~S1655-1]
openhab2-addons/unstable,now 2.5.0~S1655-1 all [installed]
```

**Comment by J-N-K** [comment]
I never install openhab2-addons, so no idea. But obviously your are not running the snapshot bindings.

**Comment by teliot** [comment]
I updated openhab and found homie persisting after reboots. Most of the channels were still dropped, but they Thing channel was now sub'ed. 

![Screenshot (3)](https://user-images.githubusercontent.com/6187302/62756544-fbf55f00-ba2c-11e9-8645-72c96334d216.png)

above before the restart you will find more topics are sub'ed. I am uncertain if the other channels are needed. 

I have tested and found this bug is fixed under latest.
